# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1250535228 14400
#      Mon Aug 17 14:53:48 2009 -0400
# Node ID 936734e15a4de7d514ebc037dacd81e631b6d5fc
# Parent  baeecd69fd44c6706bedc8423df8e71dfad2b5d3
print_usage: make it work again if an argument is given

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-08-17  John W. Eaton  <jwe@octave.org>
+
+	* help/print_usage.m: Make it work again if an argument is given.
+
 2009-08-17  Jaroslav Hajek  <highegg@gmail.com>
 
 	* help/print_usage: Allow calling from methods.
 
 2009-08-11  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/diff.m: Remove.
 	* general/Makefile.in: Update.
diff --git a/scripts/help/print_usage.m b/scripts/help/print_usage.m
--- a/scripts/help/print_usage.m
+++ b/scripts/help/print_usage.m
@@ -28,28 +28,28 @@ function print_usage (name)
   if (nargin == 0)
     ## Determine the name of the calling function
     x = dbstack ();
     if (numel (x) > 1)
       name = x (2).name;
     else
       error ("print_usage: invalid function\n");
     endif
-    path = evalin ("caller", "mfilename (""fullpath"")");
-    if (strcmp (path(end-length(name)+1:end), name))
-      path = [path, ".m"];
-    else
-      path = name;
+    fullpath = evalin ("caller", "mfilename (""fullpath"")");
+    if (strcmp (fullpath(end-length(name)+1:end), name))
+      fullname = [fullpath, ".m"];
     endif
   elseif (!ischar (name))
     error ("print_usage: input argument must be a string");
+  else
+    fullname = name;
   endif
   
   ## Do the actual work
-  [text, format] = get_help_text (path);
+  [text, format] = get_help_text (fullname);
   max_len = 80;
   switch (lower (format))
     case "plain text"
       [usage_string, status] = get_usage_plain_text (text, max_len);
     case "texinfo"
       [usage_string, status] = get_usage_texinfo (text, max_len);
     case "html"
       [usage_string, status] = get_usage_html (text, max_len);
